tapdb/universe_federation.go

```diff
@@ -328,11 +333,15 @@ func (u *UniverseFederationDB) QueryFederationSyncConfigs(
 			return err
 
 		default:
-			// Parse global db sync configs.
-			globalConfigs = make(
-				[]*universe.FedGlobalSyncConfig,
-				len(globalDbConfigs),
-			)
+			// Start with the default global sync configs.
+			for i := range defaultGlobalSyncConfigs {
```

Comment:
Is this required given that there should only actually be two global configs in the DB? IIUC, we should always override with what's in the DB.

Reply:
Also loop scoping stuff isn't needed anymore. We build+test with that new flag.

Reply:
I agree it's sort of overkill for only two types, but I figured if we plan to extended these types later on I'd rather write it this way and not have to rework it later.